tree/dataframe/inc/ROOT/RDF/RInterface.hxx

```diff
@@ -98,6 +95,13 @@ class RInterface {
    friend ColumnNames_t RDFInternal::ConvertRegexToColumns(const RNode &node,
                                                            std::string_view columnNameRegexp,
                                                            std::string_view callerName);
+   using HeadNode_t = ::ROOT::RDF::RResultPtr<RInterface<RLoopManager, void>>;
+   friend HeadNode_t RDFInternal::CreateSnaphotRDF(const ColumnNames_t &validCols,
+                                                   const std::string &fullTreeName,
+                                                   const std::string &fileName,
+                                                   bool isLazy,
+                                                   RLoopManager &loopManager,
+                                                   std::unique_ptr<RDFInternal::RActionBase> actionPtr);
```

Comment:
I think we can do without this friend declaration: if I'm not mistaken the function only needs to access the `fProxiedPtr` of a `RInterface`, but we might work around this (see below)

Reply:
Let me check this one.